### PR TITLE
[cmake] adding a target install

### PR DESCRIPTION
**Pull request by matlo607**
I propose to add a target `install` to CMakeLists.txt

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).

## Checklist:

- [x] It is my own work, its copyright is implicitly assigned to the project and no substantial part of it has been copied from other sources (including [Stack Overflow](https://stackoverflow.com/)). In rare occasions this is acceptable, like in CMake modules where the original copyright information should be kept.
- [x] I'm using the same code standards as the existing code (indentation, spacing, variable naming, ...).
- [x] I've added tests for my code.
- [x] I have verified whether my change requires changes to the documentation
- [x] My change either requires no documentation change or I've updated the documentation accordingly.
- [x] My branch has been rebased to master, keeping only relevant commits.

### CODE REVIEW

**Comment by matlo607** [comment]
@paoloambrosio, @muggenhor or @konserw : could you please review the pull request ?
Please ignore the commit https://github.com/cucumber/cucumber-cpp/pull/182/commits/3887d2cd85ea712c67aea29bc1362b78d73c88eb, konserw told me it's wrong. I will wait for #184.

**Comment by konserw** [comment]
@matlo607 could you please rebase this branch onto master now that it is fixed? Also please drop commmit
8076700

**Comment by matlo607** [comment]
Hello @konserw,
I rebased the branch and removed 8076700

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -45,9 +45,27 @@ endif()
 add_library(cucumber-cpp-nomain STATIC ${CUKE_SOURCES})
 add_library(cucumber-cpp STATIC ${CUKE_SOURCES} main.cpp)
 
+target_include_directories(cucumber-cpp-nomain PUBLIC
+    $<BUILD_INTERFACE:${CUKE_INCLUDE_DIR}>
+    $<INSTALL_INTERFACE:include>
+    PRIVATE $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR})
```
Comment (muggenhor) [inline src/CMakeLists.txt]: The current source directory isn't necessary as include path so don't add it. Same below

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -45,9 +45,27 @@ endif()
 add_library(cucumber-cpp-nomain STATIC ${CUKE_SOURCES})
 add_library(cucumber-cpp STATIC ${CUKE_SOURCES} main.cpp)
 
+target_include_directories(cucumber-cpp-nomain PUBLIC
+    $<BUILD_INTERFACE:${CUKE_INCLUDE_DIR}>
+    $<INSTALL_INTERFACE:include>
```
Comment (muggenhor) [inline src/CMakeLists.txt]: This should be `$<INSTALL_INTERFACE:$<INSTALL_PREFIX>/include>`. Same below

File: src/CMakeLists.txt [hunk 3]
```diff
@@ -45,9 +45,27 @@ endif()
 add_library(cucumber-cpp-nomain STATIC ${CUKE_SOURCES})
 add_library(cucumber-cpp STATIC ${CUKE_SOURCES} main.cpp)
 
+target_include_directories(cucumber-cpp-nomain PUBLIC
+    $<BUILD_INTERFACE:${CUKE_INCLUDE_DIR}>
```
Comment (muggenhor) [inline src/CMakeLists.txt]: Is adding this enough to permit us from removing the directory-scoped `include_directories(${CUKE_INCLUDE_DIR})` in the toplevel `CMakeLists.txt`?

File: src/CMakeLists.txt [hunk 4]
```diff
@@ -45,9 +45,27 @@ endif()
...
 endif(MINGW)
 
 target_link_libraries(cucumber-cpp-nomain PRIVATE ${CUKE_DEP_LIBRARIES})
 target_link_libraries(cucumber-cpp PRIVATE ${CUKE_DEP_LIBRARIES})
+
+install(TARGETS cucumber-cpp-nomain
+    ARCHIVE  DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    LIBRARY  DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    RUNTIME  DESTINATION ${CMAKE_INSTALL_BINDIR})
+install(TARGETS cucumber-cpp
```
Comment (muggenhor) [inline src/CMakeLists.txt]: Why the duplication? The `install()` command supports multiple targets after `TARGETS`.

File: src/CMakeLists.txt [hunk 5]
```diff
@@ -45,9 +45,27 @@ endif()
...
+    ARCHIVE  DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    LIBRARY  DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    RUNTIME  DESTINATION ${CMAKE_INSTALL_BINDIR})
```
Comment (muggenhor) [inline src/CMakeLists.txt]: For usability I'm expecting an `export()` of the library targets too.

**Comment by aslakhellesoy** [comment]
Hi @matlo607,

Thanks for your making your first contribution to Cucumber, and welcome to the Cucumber committers team! You can now push directly to this repo and all other repos under the cucumber organization! 🍾

In return for this generous offer we hope you will:

* ✅ Continue to use branches and pull requests. When someone on the core team approves a pull request (yours or someone else's), you're welcome to merge it yourself.
* 💚 Commit to setting a good example by following and upholding our [code of conduct](https://cucumber.io/conduct) in your interactions with other collaborators and users.
* 💬 Join the [community Slack channel](http://cucumber.io/support#slack) to meet the rest of the team and make yourself at home.
* ℹ️ Don't feel obliged to help, just do what you can if you have the time and the energy.
* 🙋 Ask if you need anything. We're looking for feedback about how to make the project more welcoming, so please tell us!

On behalf of the Cucumber core team,
Aslak Hellesøy
Creator of Cucumber

**Pull request merged**